formatter/src/main/java/org/togetherjava/formatter/Formatter.java

```diff
@@ -0,0 +1,319 @@
+    }
+
+    /**
+     * Writes and formats a given code section (in form of a list of tokens) into a StringBuilder
+     *
+     * @param tokens tokens to write
+     * @return written code sections
+     * @author illuminator3
+     */
+    private StringBuilder writeCodeSection(List<Token> tokens) {
+        purgeWhitespaces(tokens = makeMutable(tokens));
+
+        TokenType lastToken = TokenType.UNKNOWN;
+        StringBuilder result = new StringBuilder();
+        boolean lastNewLine = false;
+        int indentation = 0;
+        boolean forClosed = true;
+        int forClosingLevel = 0;
+
+        for (Token token : tokens) {
+            TokenType type = token.type();
+
+            if (lastNewLine && type != TokenType.CLOSE_BRACES) {
+                putIndentation(indentation, result);
+            }
+
+            lastNewLine = false;
+
+            iteration: {
+                if (type == TokenType.OPEN_BRACES) {
+                    if (lastToken == TokenType.CLOSE_PARENTHESIS
+                            || lastToken == TokenType.IDENTIFIER || lastToken == TokenType.TRY || lastToken == TokenType.ELSE) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+                    indentation++;
+
+                    break iteration;
+                } else if (lastToken == TokenType.CLOSE_BRACKETS && type != TokenType.SEMICOLON) {
+                    result.append(" ");
+
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.CLOSE_BRACES) {
+                    result.append("\n");
+
+                    putIndentation(--indentation, result);
+
+                    append(result, token);
+
+                    result.append("\n");
+
+                    lastNewLine = true;
+
+                    break iteration;
+                } else if (type == TokenType.IDENTIFIER && (lastToken == TokenType.CLOSE_BRACES
+                        || lastToken == TokenType.OPEN_BRACES
+                        || lastToken == TokenType.SEMICOLON
+                        || lastToken == TokenType.UNKNOWN)) {
+                    append(result, token);
+
+                    break iteration;
+                } else if (type == TokenType.COMMENT || type == TokenType.SEMICOLON
+                        || type == TokenType.ANNOTATION) {
+                    if (type == TokenType.COMMENT) {
+                        result.append(" ");
+                    }
+
+                    append(result, token);
+
+                    if (forClosed) {
+                        result.append("\n");
+
+                        lastNewLine = true;
+                    }
+
+                    break iteration;
+                } else if ((type == TokenType.OPEN_PARENTHESIS && lastToken != TokenType.IDENTIFIER
```

Comment:
This is *definitely* too much for an if. As above, turn this into one (or more!) *aptly named* function calls describing what it does.